internal/dim-2d/dim-2d.go

```diff
@@ -0,0 +1,120 @@
+// Package dim2d implements the 2D sub algorithm
+package dim2d
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/edaniels/golog"
+	"github.com/pkg/errors"
+	"go.opencensus.io/trace"
+	"go.viam.com/rdk/registry"
+	"go.viam.com/slam/dataprocess"
+	"go.viam.com/slam/sensors/lidar"
+	goutils "go.viam.com/utils"
+)
+
+const (
+	opTimeoutErrorMessage = "bad scan: OpTimeout"
+)
+
+// NewLidar returns a new lidar.Lidar for the 2D cartographer mode.
+func NewLidar(
+	ctx context.Context,
+	deps registry.Dependencies,
+	sensors []string,
+	logger golog.Logger,
+) (lidar.Lidar, error) {
+	_, span := trace.StartSpan(ctx, "viamcartographer::internal::dim2d::NewLidar")
+	defer span.End()
+
+	// An empty `sensors: []` array is allowed in offline mode.
+	if len(sensors) == 0 {
+		logger.Debug("no sensor provided in 'sensors' config parameter")
+		return lidar.Lidar{}, nil
+	}
+	// If there is a sensor provided in the 'sensors' array, we enforce that only one
+	// sensor has to be provided.
+	if len(sensors) != 1 {
+		return lidar.Lidar{}, errors.Errorf("configuring lidar camera error: "+
+			"'sensors' must contain only one lidar camera, but is 'sensors: [%v]'",
+			strings.Join(sensors, ", "))
+	}
+
+	lidar, err := lidar.New(deps, sensors, lidar2DIndex)
+	if err != nil {
+		return lidar, errors.Wrap(err, "configuring lidar camera error")
+	}
+
+	return lidar, nil
+}
+
+// ValidateGetAndSaveData makes sure that the provided sensor is actually a lidar and can
```

Comment:
[q] Should the bring-up of a lidar be held in dim-2d as future work in cartographer to support 3D lidars would duplicate much of this. For example the `ValidateGetAndSaveData` would be identical as would the `GetAndSaveData`.